### ISSUE TITLE
Instructions for oAuth

### ISSUE THREAD

**Issue by ohabash** [getconversio/node-bigcommerce#50]
Thank you in advance! I am struggling here.
 
Can you please provide complete instructions on how my users can authenticate and receive their token. There are scattered instructions between your readme and the BigCommerce docs but the exact process with nodejs is very cryptic. Any help on this is appreciated.

**Comment by dnapoleoni** [comment]
Agreed!

**Comment by dnapoleoni** [comment]
That over-explains it a lot and gives a lot of examples that aren't in node.js, it'd be great to get an example of the process that:
a) describes how it integrates with the BigCommerce ecosystem
b) shows how to use your package to create a BigCommerce app

I also found the examples that BC provide overwhelming and the examples in your readme seem to be missing a lot of info.

**Comment by ohabash** [comment]
@stefanosala ur module is working great for me now that i understand what I need to do better.

@dnapoleoni 
I was able to get this working. I had 3 misunderstandings that may help. Refer to [bc api docs](https://developer.bigcommerce.com/api) for specifics. 

1. BigCommerce does not have api authentication for individual users only merchants (store owners) or merchant apps on behalf of the user. Then as a merchant you have access to actions on behalf of your users. You can use the [customer api](https://developer.bigcommerce.com/api/v3/customers.html) to look them up by several parameters.

2. I was trying to call the auth method manually resulting in the "redirecting to login" message. Actually, the auth method is NOT called manually by us - bigCommerce will call [that method](https://developer.bigcommerce.com/api/#receiving-the-get-request) (at ur defined callback setup during [app creation](https://developer.bigcommerce.com/api/#registering-your-app) in the developers portal). When that happens you will start oAuth - which looks like the following. 
- a merchant initializes an app in their store. for example I made an app for my store so I need to give my app permission to do bc stuff on my behalf. (toggle permissions of how to limit the reach of a particular app)
- bigCommerce will send a code to ur callback
- in the callback send that same code right back to bigCommerce and the response from bigcommerce will be ur merchant token which can be used to access the api endpoints.

3. you can skip step 2 during dev by [creating an api account from the bc merchants' admin](https://developer.bigcommerce.com/api/v3/customers.html). then just hard code the given token and skip the above until u [register](https://developer.bigcommerce.com/api/#registering-your-app) and [install](https://developer.bigcommerce.com/api/#create-your-draft-app) it properly on bc merchant admin.

Feel free to correct any of this or ask follow up.

**Comment by dnapoleoni** [comment]
Hey @stefanosala, thanks for the reply.

I've checked out both of those guides, but still seems like I'm most of the way there but a few of the finer details I can't quite work out the exact implementation.

In the code above I'm storing the accessToken in the config.bigCommerce object (inside of the /auth call), so I assumed that when I create the new bigCommerce object and pass the object in that it would come with the accessToken included. According to a log at the /hooks call level it seems like it's not passing through, maybe there was somewhere else I was meant to be storing that variable on receiving it (in /auth) and passing it back (in /hooks)?

Either that or i need to work out how to uninstall the app, because maybe the changes I'm making to the later calls are being refreshed, but the original /auth code doesn't get called once it's been installed so it's never running the code that I added later.

Not sure! Will do more testing and work it out. I've changed the api version to 'v2' and the '/hooks' URI to your suggested method (by removing the full url) and the issue is still there so it must be something else.

Cheers!
